thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java

```diff
@@ -375,6 +398,43 @@ void updateDetectionPipeline(long detectionID, String payload, long startTime, l
     }
   }
 
+  private boolean isServiceAccount(ThirdEyePrincipal user) {
+    List<Predicate> predicates = new ArrayList<>();
+    predicates.add(Predicate.EQ("sessionKey", user.getSessionKey()));
+    predicates.add(Predicate.EQ("principalType", "SERVICE"));
+
+    List<SessionDTO> sessionDTO = this.sessionDAO.findByPredicate(Predicate.AND(predicates.toArray(new Predicate[0])));
+    return sessionDTO != null && !sessionDTO.isEmpty();
```

Comment:
Do we check the session expiration time?

Reply:
Yes, session expiry is checked separately by the auth module.

Reply:
make sense